### PR TITLE
Add an inital ROADMAP for the operator

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

To be completed and discussed during next working group(s).
Feel free to comment any other ideas too 👼🏼 

/assign @tektoncd/operator-maintainers 
/cc @tektoncd/operator-collaborators 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- custom tasks
+- other projects
+
+## More targets
+
+We are currently targeting and releasing only two target:
+- Vanilla k8s
+- OpenShift
+
+We should aim to support more, starting with GKE. GKE is a easy target
```
Comment (ImJasonH) [inline ROADMAP.md:39]: Do we have an idea whether/how GKE will differ from Vanilla K8s?

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Towards seamless upgrades
+
+TBD
+
+## Pipeline
+
+### Support feature flags
+
+TBD
```
Comment (ImJasonH) [inline ROADMAP.md:11]: Can we list some examples of things an operator might enable/disable?

- Affinity Assistant
- Custom Tasks
- Maybe disabling ClusterTasks?

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+### Support configuration options
+
+TBD
```
Comment (ImJasonH) [inline ROADMAP.md:15]: Enabling/configuring HA correctly?

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+## New Component integration
+
+As of today, the operator is capable of installing Pipeline, Triggers
+and the dasbhoard. We may want to support shipping more components
+
+### Components
+
+- results
+- chains
+- hub self-hosted
```
Comment (savitaashture) [inline ROADMAP.md:26]: we can also enable creation of clusterTask as part of Operator Installation

File: ROADMAP.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+## Towards seamless upgrades
+
+TBD
+
+## Pipeline
+
+### Support feature flags
+
+TBD
+
```
Comment (savitaashture) [inline ROADMAP.md:12]: Similar to `Pipeline` we can have section for `Triggers` also
As Triggers supports 
* custom interceptor and there is a plan to install builtin interceptor by admin(operator) cc @dibyom correct me if miss something
* Multi tenant EL support cc @khrm please add if anything we need to handle from Operator side

File: ROADMAP.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Roadmap
+
+## Towards seamless upgrades
+
+TBD
+
```
Comment (savitaashture) [inline ROADMAP.md:6]: We can target for enabling metrics for Operator

File: ROADMAP.md (hunk 5 above)
Comment (dibyom) [reply ROADMAP.md:12]: > custom interceptor and there is a plan to install builtin interceptor by admin(operator) cc @dibyom correct me if miss something

right now the plan is to bundle the 4 interceptors as we do today but of course a downstream admin can decide to add/remove/change the set of interceptors they ship.

**Comment by vdemeester** [comment]
Updated 👼🏼 
/cc @tektoncd/operator-maintainers

**Review by savitaashture — APPROVED** [review]

**Review by pradeepitm12 — APPROVED** [review]
/lgtm

**Pull request merged**